sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/common/ClientConfiguration.java

```diff
@@ -132,58 +156,8 @@ public Builder retry(Consumer<RetryConfiguration.Builder> retry) {
 
     abstract Builder regionId(String region);
 
-    abstract AwsCredentialsProvider credentialsProvider();
-
-    abstract ClientConfiguration autoBuild();
+    abstract Builder credentialsProviderAsJson(String credentialsProvider);
 
-    public ClientConfiguration build() {
-      if (credentialsProvider() != null) {
-        credentialsProvider(new SerializableAwsCredentialsProvider(credentialsProvider()));
-      }
-      return autoBuild();
-    }
-  }
-
-  /** Internal serializable {@link AwsCredentialsProvider}. */
-  private static class SerializableAwsCredentialsProvider
```

Comment:
Removing SerializableAwsCredentialsProvider to simplify Json serialization, otherwise this would require a custom serializer (and can't be private any longer). Instead ClientConfiguration just stores the serialized Json representation internally.